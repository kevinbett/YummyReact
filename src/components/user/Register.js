import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import axios from 'axios';
import '../../static/css/home.css';
import {notify} from 'react-notify-toast';


class Register extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            email: '',
            password: '',
        }
    }
    render() {
        return (

            <div className="Register">
                <MuiThemeProvider>
                    <div>
                        <div>
                        <form>
                            <TextField
                                hintText="Enter your Username"
                                floatingLabelText="Username"
                                onChange={(event, newValue) =>
                                    this.setState({ username: newValue })} />
                            <br />
                            <TextField
                                hintText="Enter your Email"
                                floatingLabelText="Email"
                                onChange={(event, newValue) =>
                                    this.setState({ email: newValue })} />
                            <br />
                            <TextField
                                type="password"
                                hintText="Enter your Password"
                                floatingLabelText="Password"
                                onChange={(event, newValue) =>
                                    this.setState({ password: newValue })} />
                            <br />
                            <RaisedButton label="Submit" secondary={true}
                                style={style} onClick={(event) => this.handleClick(event)} />
                            </form>
                        </div>
                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
    handleClick(event) {
        var apiBaseUrl = "http://localhost:5000/api/auth/";
        var payload = {
            "username": this.state.username,
            "password": this.state.password,
            "email": this.state.email
        }
        axios.post(apiBaseUrl + 'register/', payload)
            .then(response => {
                setTimeout(function(){ window.location.assign(`/login`) }, 2000);
                notify.show(response.data.message, 'success', 4000)                               
            })
            .catch(error => {
                if (error.response.data.message.error) {
                    notify.show(error.response.data.message.error, 'error', 4000);                    
                } else if (error.response.data.message.email) {
                   notify.show(error.response.data.message.email[0], 'error', 4000)
                }
            })
    }

}

const style = {
    marginRight: 60,
    float: 'center'
};
export default Register;