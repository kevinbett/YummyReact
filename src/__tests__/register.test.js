import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Register from '../login/register';

describe('Register Component', () => {
    const wrapper = shallow(<Register />);
    it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('renders state initially', () => {
        expect(wrapper.state().username).toEqual('');
        expect(wrapper.state().email).toEqual('');        
        expect(wrapper.state().password).toEqual('');
    })
});