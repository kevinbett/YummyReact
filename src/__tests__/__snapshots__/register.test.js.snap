// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component renders without crashing 1`] = `
<div
  className="Register"
>
  <MuiThemeProvider>
    <div>
      <div>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Username"
          fullWidth={false}
          hintText="Enter your Username"
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Email"
          fullWidth={false}
          hintText="Enter your Email"
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Password"
          fullWidth={false}
          hintText="Enter your Password"
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="password"
          underlineShow={true}
        />
        <br />
        <div>
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Submit"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={true}
            style={
              Object {
                "marginRight": 20,
              }
            }
          />
        </div>
      </div>
    </div>
  </MuiThemeProvider>
</div>
`;
